Formula/biosig.rb

```diff
@@ -0,0 +1,31 @@
+class Biosig < Formula
+  desc "Tools for biomedical signal processing and data conversion"
+  homepage "https://biosig.sourceforge.io"
+  url "https://downloads.sourceforge.net/project/biosig/BioSig%20for%20C_C%2B%2B/src/biosig4c%2B%2B-1.9.5.src.tar.gz"
+  sha256 "20e72a5a07d1bf8baa649efe437b4d3ed99944f0e4dfc1fbe23bfbe4d9749ed5"
+
+  depends_on "gawk" => :build
+  depends_on "gnu-sed" => :build
+  depends_on "gnu-tar" => :build
+  depends_on "pkg-config" => :build
+  depends_on "dcmtk"
+  depends_on "libb64"
+  depends_on "suite-sparse"
+  depends_on "tinyxml"
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    assert_match "usage: save2gdf [OPTIONS] SOURCE DEST", shell_output("#{bin}/save2gdf -h").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I've added such a test case. The input has to be a binary file, since I do not know how to add such a binary test file in the brew repository, the current test downloads a test file from my homepage.